pkg/flowaggregator/flowaggregator.go

```diff
@@ -194,7 +275,28 @@ func (fa *flowAggregator) Run(stopCh <-chan struct{}) {
 					continue
 				}
 			}
-			err := fa.aggregationProcess.ForAllRecordsDo(fa.sendFlowKeyRecords)
+			dataSet := ipfix.NewSet(ipfixentities.Data, fa.templateID, false)
+			addFlowKeyRecord := func(key ipfixintermediate.FlowKey, record ipfixintermediate.AggregationFlowRecord) error {
+				if !record.ReadyToSend {
+					klog.V(4).Info("Skip sending record that is not correlated.")
+					return nil
+				}
+				// limit dataSet to have no more than two records
+				// TODO: more records per dataSet will be supported when go-ipfix supports size check when adding records
+				if dataSet.GetSet().GetNumberOfRecords() == recordNumPerSet {
+					_, err := fa.sendDataSet(dataSet)
+					if err != nil {
+						return err
+					}
+					dataSet = ipfix.NewSet(ipfixentities.Data, fa.templateID, false)
+				}
```

Comment:
I don't understand why this has to be exposed by the library? Can't the library automatically send a "set" when appropriate?

Also, the comment doesn't explain to me why the number of records has to be 2 - the fact that we need that kind of logic doesn't give me a lot of confidence in the software TBH

Reply:
Thanks for the comment, Antonin.

Prior to v0.4.0, we had an interface AddSetAndSendMsg, where we were maintaining the Set object inside the exporter process struct. However, in the process of simplifying things we moved to the SendSet interface and let the user provide the data set. Yes, we saw some issues in sending multiple records in one set and creating complexity on the user side.  
We should fall back to AddSetAndSendMsg and manage the sets as part of the exporter process struct in the library.

@zyiou To avoid confusion, let us just send one record per set now by creating data set inside the callback function.  Thanks for taking care of this.

Reply:
Currently we don't support dynamic message size check when adding record to set. We only have size check when sending the set because the size of the whole data to be sent will also include the set header, message header etc. We plan to support it in the future. 
I agree 2 is just a subjectively safe estimation and we need to change the root method in go-ipfix instead. For this PR I will change it back to 1 record per set until we have support from go-ipfix. Thanks!

Reply:
Yes please set it to 1 to avoid confusion, as I have no idea what "dynamic message size check" is. Thanks @srikartati and @zyiou